python3/electro_grammar/electro_grammar.py

```diff
@@ -183,6 +183,39 @@ def exitRtype(self, ctx):
         if ctx.POT():
             self.obj['rtype'] = 'pot'
 
+    def exitDiode(self, ctx):
+        self.obj['type'] = 'diode'
+
+    def exitDcode(self, ctx):
+        self.obj['code'] = ctx.DCODE().getText().upper()
+
+    def exitSignal(self, ctx):
+        self.obj['dtype'] = 'signal'
+
+    def exitRectifier(self, ctx):
+        self.obj['dtype'] = 'rectifier'
+
+    def exitLed(self, ctx):
+        self.obj['dtype'] = 'led'
+
+    def exitSchottky(self, ctx):
+        self.obj['dtype'] = 'schottky'
```

Comment:
Could we use more verbose names? `diode_type` instead of `dtype`. `transistor_type` instead of `ttype`.